### PR TITLE
bump @material-ui/core & icon to ^4

### PR DESCRIPTION
**Pull request by MrSpoocy**
simple bump to higher version to use some new future and not develop on old version.

### CODE REVIEW

**Comment by benzino77** [comment]
How about `rows` and `rowsMax` properties in MessageBox.js?

**Comment by benzino77** [comment]
I'm not sure if this is only me but fonts looks "bigger" then before.....

**Comment by benzino77** [comment]
https://mui.com/guides/migration-v3/#typography 
That's why I had a feeling that fonts are bigger ;)

**Comment by MrSpoocy** [comment]
Question is ... we want to change the font size back or go for the default ?

(currently I add the future that we have a sub menu e.g. if you check "Use MQTT" you will get sub steps on "Custom Paramater" to set username, passsword, host ...). Without add extra components. Simple use nested entry in AvailableFutures.js

File: src/App.js [hunk 1]
```diff
@@ -135,13 +135,13 @@ class App extends Component {
 
   changeLanguage = (lang) => {
     tasmotaGUILanguages.sort((a, b) => {
-      return allMessages[lang]['source'][a.name].localeCompare(
-        allMessages[lang]['source'][b.name]
+      return (allMessages[lang]['source'][a.name] || a.name).localeCompare(
+          (allMessages[lang]['source'][b.name] || b.name)
```
Comment (benzino77) [inline src/App.js:139]: I don't think we need to do this in the code of the application. What needs to be implemented are proper test scenarios to find out whether there is a translation with a missing key.
`npm run test` - that is needed. This should tell us that some translations are incomplete.

Have a look at my conversation with @drhideg here: #197

**Comment by MrSpoocy** [comment]
@benzino77 I'm not sure if I'm blind ...bind can not find a Button for sending a private message to you xD

So I will ask you here (sorry for that), is there a reason why you use filter instand of find in many cases? When I add the new future, I also add support for TypeScript (because I write everything in TS). I have setting the compiler to ES2020. I see that you use  

```
 "browserslist": [
    ">0.2%",
    "not dead",
    "not ie <= 11",
    "not op_mini all"
  ],
```

You exclude IE <= 11 that means its save to use ES2020 (only some not used feature not supported)

**Comment by benzino77** [comment]
>  is there a reason why you use filter instead of find in many cases?

No. Really. There is no particular reason for that. I don't even realize that I like `filter` more than `find` :)

`browserslist` was generated by `create-react-app` - I didn't touch it since then:
git blame shows this information:

image

**Comment by benzino77** [comment]
I've created quick and dirty font size workaround:
in App.js add this at the top of the file
```javascript
const tcTheme = createTheme({
  typography: {
    body1: {
      fontSize: '0.875rem', 
      lineHeight: 1.43, 
      letterSpacing: '0.01071em',
    }
  }
})

const browserLanguage = navigator.language.toLocaleLowerCase();
```

then wrap everything with `ThemeProvider`:
```javascript
...
return (
      <IntlProvider
        locale={tcGUILanguage}
        messages={allMessages[tcGUILanguage]['source']}
      >
        <div className={classes.root}>
          <ThemeProvider theme={tcTheme}>
          <TopAppBar
...
```
but I'm not sure that this is the best solution....
Maybe the font size of step name/step label should be bigger?

**Comment by MrSpoocy** [comment]
Maybe wrap StepLabel in HOC with own font size.

File: src/App.js (hunk 1 above)
Comment (benzino77) [reply src/App.js:139]: Have a look here: https://github.com/benzino77/tasmocompiler/blob/da5c38436aa36d6ff4677d9121a44e391d5f506d/src/locales/languages.js#L71-L78

I've already addressed the situation that someone has forgot about some translation. Completely forget about that ;)

**Comment by MrSpoocy** [comment]
I still try to bump all dependencies ... and try to use your eslint settings ... not sure if you really used it xD So many warning/errors that i get. Now **many** of them already fixed, feel free to check it one time


https://github.com/MrSpoocy/tasmocompiler/tree/bump_dependencies

**Comment by benzino77** [comment]
There was a moment I've changed my workstation... and for code formatting I started using Prettier. Probably I had forgotten about eslint, to keep the code align with rules. I think it should be used.

Let't not make this PR too big. This one should only be related to bumbing material-ui version. I will try to merge this part this week. 

For updating React, code formatting/best practices with eslint should be directed to new PRs.

PS.
Manuel thanks for your contribution. For a long time, no one has put the effort to contribute to TasmoCompiler.

**Pull request merged**